test/src/tests.js

```diff
@@ -1593,4 +1718,222 @@ describe('Rokt Forwarder', () => {
+                filterUserAttributes: function (attributes) {
+                    return attributes;
+                },
+                filteredUser: {
+                    getMPID: function () {
+                        return '123';
+                    },
+                },
+            };
+        });
+
+        it('set a session selection attribute if the event is a mapped placement event', async () => {
+            // Mocks hashed values for testing
+            const placementEventMapping = JSON.stringify([
+                {
+                    jsmap: 'hashed-<48Video Watched>-value',
+                    map: '123466',
+                    maptype: 'EventClass.Id',
+                    value: 'foo-mapped-flag',
+                },
+                {
+                    jsmap: 'hashed-<29Other Value>-value',
+                    map: '1279898989',
+                    maptype: 'EventClass.Id',
+                    value: 'ad_viewed_test',
+                },
+            ]);
+
+            await window.mParticle.forwarder.init(
+                {
+                    accountId: '123456',
+                    placementEventMapping,
+                },
+                reportService.cb,
+                true,
+                null,
+                {}
+            );
+
+            await waitForCondition(() => window.mParticle.Rokt.attachKitCalled);
+
+            window.mParticle.forwarder.processEvent({
+                EventName: 'Video Watched',
+                EventCategory: EventType.Other,
+                EventDataType: MessageType.PageEvent,
+            });
+
+            window.mParticle._Store.localSessionAttributes.should.deepEqual({
+                'foo-mapped-flag': true,
+            });
+        });
+    });
+
+    describe('#parseSettingsString', () => {
+        it('should parse a settings string', () => {
+            const settingsString =
+                '[{&quot;jsmap&quot;:null,&quot;map&quot;:&quot;f.name&quot;,&quot;maptype&quot;:&quot;UserAttributeClass.Name&quot;,&quot;value&quot;:&quot;firstname&quot;},{&quot;jsmap&quot;:null,&quot;map&quot;:&quot;last_name&quot;,&quot;maptype&quot;:&quot;UserAttributeClass.Name&quot;,&quot;value&quot;:&quot;lastname&quot;}]';
+
+            window.mParticle.forwarder.testHelpers
+                .parseSettingsString(settingsString)
+                .should.deepEqual([
+                    {
+                        jsmap: null,
+                        map: 'f.name',
+                        maptype: 'UserAttributeClass.Name',
+                        value: 'firstname',
+                    },
+                    {
+                        jsmap: null,
+                        map: 'last_name',
+                        maptype: 'UserAttributeClass.Name',
+                        value: 'lastname',
+                    },
+                ]);
+        });
+
+        it('should parse a settings string with a stringified number value correctly', () => {
```

Comment:
I'm still unclear about the meaning of this test.  what about a stringified number is being tested?  saw that you resolved the original comment but didn't see any changes

Reply:
The purpose of this test is to verify that if a setting string is parsed, the string hashes in the `jsmap` and `map` properties are not converted to numbers, but remain as strings.

For example, `&quot;jsmap&quot;:&quot;-1484452948&quot;` should be come `jsmap: '-1484452948'` (sting) and not `-1484452948` (number)